numpy/core/getlimits.py

```diff
@@ -8,20 +8,239 @@
+                         huge=_huge_f80,
+                         tiny=_tiny_f80)
+
+# Guessed / known parameters for double double; see:
+# https://en.wikipedia.org/wiki/Quadruple-precision_floating-point_format#Double-double_arithmetic
+# These numbers have the same exponent range as float64, but extended number of
+# digits in the significand.
+_huge_dd = (umath.nextafter(_ld(inf), _ld(0))
+            if hasattr(umath, 'nextafter')  # Missing on some platforms?
+            else _ld(inf))
+_float_dd_ma = MachArLike(_ld,
+                          machep=-105,
+                          negep=-106,
+                          minexp=-1022,
+                          maxexp=1024,
+                          it=105,
+                          iexp=11,
+                          ibeta=2,
+                          irnd=5,
+                          ngrd=0,
+                          eps=exp2(_ld(-105)),
+                          epsneg= exp2(_ld(-106)),
+                          huge=_huge_dd,
+                          tiny=exp2(_ld(-1022)))
+
+
+# Key to identify the floating point type.  Key is result of
+# ftype('-0.1').newbyteorder('<').tobytes()
+# See:
+# https://perl5.git.perl.org/perl.git/blob/3118d7d684b56cbeb702af874f4326683c45f045:/Configure
+_KNOWN_TYPES = {
+    b'\x9a\x99\x99\x99\x99\x99\xb9\xbf' : _float64_ma,
+    b'\xcd\xcc\xcc\xbd' : _float32_ma,
+    b'f\xae' : _float16_ma,
+    # float80, first 10 bytes containing actual storage
+    b'\xcd\xcc\xcc\xcc\xcc\xcc\xcc\xcc\xfb\xbf' : _float80_ma,
+    # double double; low, high order (e.g. PPC 64)
+    b'\x9a\x99\x99\x99\x99\x99Y<\x9a\x99\x99\x99\x99\x99\xb9\xbf' :
+    _float_dd_ma,
+    # double double; high, low order (e.g. PPC 64 le)
+    b'\x9a\x99\x99\x99\x99\x99\xb9\xbf\x9a\x99\x99\x99\x99\x99Y<' :
+    _float_dd_ma,
+}
+
+
+def _get_machar(ftype):
+    """ Get MachAr instance or MachAr-like instance
+
+    Get parameters for floating point type, by first trying signatures of
+    various known floating point types, then, if none match, attempting to
+    identify parameters by analysis.
+
+    Parameters
+    ----------
+    ftype : class
+        Numpy floating point type class (e.g. ``np.float64``)
+
+    Returns
+    -------
+    ma_like : instance of :class:`MachAr` or :class:`MachArLike`
+        Object giving floating point parameters for `ftype`.
+    """
+    params = _MACHAR_PARAMS.get(ftype)
+    if params is None:
+        raise ValueError(repr(ftype))
+    # Detect known / suspected types
+    key = ftype('-0.1').newbyteorder('<').tobytes()
+    if key in _KNOWN_TYPES:
+        return _KNOWN_TYPES[key]
+    # Could be 80 bit == 10 byte extended precision, where last bytes can be
+    # random garbage.  Try comparing first 10 bytes to pattern.
+    if ftype == ntypes.longdouble:
+        key = key[:10]
+        if key in _KNOWN_TYPES:
+            return _KNOWN_TYPES[key]
+    # Fall back to parameter discovery
+    return _discovered_machar(ftype)
```

Comment:
Might be worth raising a warning here so that we might get reports of new types we should cover.